src/wbi/legacy/flash_finder.py

```diff
@@ -1,63 +1,213 @@
-import numpy as np
+# Copied from
```

Comment:
I feel like at this stage, it is worthwhile to still keep the original legacy files we got from Junang in the `legacy` folder, with minimal changes. Changes to make stuff work with python 3.10, as well as tweaks to test it faster are fine, but we need not worry about pre-commit/pyright complaints (these can be fixed by telling these tools to ignore the `legacy` folder).

To that effect, I have reintroduced the `flash_finder.py` file that we currently have in `main` right here. It's the exact same file.